doc/developers/advanced_installation.rst

```diff
@@ -201,6 +201,88 @@ It is however preferred to use pip.
 On Unix-like systems, you can equivalently type ``make in`` from the top-level
 folder. Have a look at the ``Makefile`` for additional utilities.
 
+.. _building_with_meson:
+
+Building with Meson
+-------------------
+
+Supported for Meson is experimental, in scikit-learn 1.5.0.dev0.
+`Open an issue <https://github.com/scikit-learn/scikit-learn/issues/new>`__ if
+you encounter any problems!
+
+Make sure you have `meson-python` installed:
+
+.. code-block:: bash
+
+    conda install -c conda-forge meson-python -y
+
+.. warning::
+
+    Use conda for this, if you install meson-python with pip, this does not not
+    include ninja, and you need another way to install ninja (e.g. package
+    manager)
```

Comment:
This is gonna be a challenge. We have a LOT of contributors who don't use mamba/conda. But I think it'd be okay, yet another system dependency like the compiler. I'm just not sure about ninja version compatibility with meson-python.

Reply:
This note should be removed I think. You can `pip install ninja` just fine (even though it's better done through you distro package manager perhaps).

Reply:
Done, I was misled by the fact that `pip install meson-python` did not install `ninja`. Maybe there is a good reason for this?

Reply:
Because ninja is a C++ application, not a python application.

It is more common for ninja to be installed by a linux distro or as part of Visual Studio, than it is for ninja to be installed from (the unofficial third-party package on) PyPI. In such cases, it won't be detected as "installed" by pip at all, no more than GNU make is detected as installed, or GNU bash.

What meson-python does is check at the time you invoke the build-backend, whether `ninja` is available somewhere on $PATH. If it is NOT available, then meson-python adds ninja to get_requires_for_build_wheel, so pip installs ninja *after* installing and running meson-python, but before launching `meson setup`.

So, meson-python guarantees that ninja is installed. But it doesn't voice an opinion on whether you install it via your package manager or via pip install.

Reply:
OK good to know, thanks a lot for the details!

Because we use `--no-build-isolation`, the mechanism for installing `ninja` automatically is not going to cover our use case unfortunately.

This is really not a big deal at all, this is completely fine to tell contributors to do `pip install meson-python ninja`.

Reply:
Indeed, absolutely. And if the contributors are positive they already have ninja, they can modify the command line they run, which is quite trivial as long as meson-python doesn't unconditionally force it.